OrbitGl/OpenGlDetect.cpp

```diff
@@ -0,0 +1,68 @@
+#include "OpenGlDetect.h"
+
+#include <absl/strings/str_split.h>
+
+#include <charconv>
+
+#include "OpenGl.h"
+#include "OrbitBase/Logging.h"
+
+namespace OrbitGl {
+
+std::optional<OpenGlVersion> DetectOpenGlVersion(GlutContext*) {
+  const auto window =
+      glutCreateWindow("Determining supported OpenGL version...");
+
+  if (window <= 0) {
+    return std::nullopt;
+  }
+
+  glewInit();
+
+  const auto str_ptr = glGetString(GL_VERSION);
+  std::string str{};
+
+  if (str_ptr != nullptr) {
+    str = std::string{reinterpret_cast<const char*>(str_ptr)};
+  }
+
+  glutDestroyWindow(window);
+
+  if (str_ptr == nullptr) {
+    return std::nullopt;
+  }
+
+  // We have to trigger the event loop three times to destroy the window fully.
+  glutMainLoopEvent();
+  glutMainLoopEvent();
+  glutMainLoopEvent();
+
+  std::vector<std::string> pieces =
```

Comment:
nit. "tokens" is usually used in the codebase instead of "pieces"

Reply:
Ah sorry. I didn't expect you to have a close-up look this night. I will make a follow-up PR.